app/assets/javascripts/angular/directives/graph_chart.js

```diff
@@ -54,7 +54,10 @@ angular.module("Prometheus.directives").directive('graphChart', ["$location", "W
         var graphHeight = WidgetHeightCalculator(element[0], scope.aspectRatio);
         $el.css('height', graphHeight);
 
+        var seriesState = {};
         if (rsGraph) {
+          // Create a unique name using the series labels to store legend state for newly drawn graph.
+          rsGraph.series.forEach(function(s) { seriesState[renderLabels(s.labels)] = s.disabled });
```

Comment:
Could we optimize for the common case for how much state we need to keep? For example, only save disabled items in this hash, which should normally be 0? You could even make it more efficient (assuming that usually people will select exactly one timeseries if they do select one) by not saving anything if everything is enabled, and only saving the enabled ones otherwise.

Also, I wouldn't base the identification of a timeseries based on `renderLabels`. It doesn't include the metric name, but it does include superfluous stuff like HTML. The best identification would be `metric_name{label1="value1", ...}` or something like that. Basically, the default name we give to a timeseries in the legend if there is no format string configured.